source/develop/release-management/features/storage/incremental-backup.html.md

```diff
@@ -12,173 +12,421 @@ feature_status: Design
+#### Use case 1 - running 2 backup solutions in the same time
+
+User is switching from backup solution A to B. During the transition,
+both systems are performing backups of the same data. If one of the
+systems will delete checkpoints after backup, it can prevent incremental
+backup of the other system.
+
+#### Use case 2 - running backups with different schedule
+
+User is configuring both hourly and daily backup. If both jobs delete
+the checkpoints after backup they may prevent the next job from
+completing the next backup.
+
+### Cancelling backup
+
+When backup fail or aborted, backup application did completed to copy
+all data.
+
+1. Delete scratch disks created for this backup.
+2. If the VM is not running, tear down the disks included in the backup.
+
+
+### Engine database changes
+
+XXX Add disk attachment incremental setting.
+
+- New ```vm_backups``` table
+  - fields: id, vm_id, incremental, creation_date, status
+  - status: initializing, ready, transferring, finallizing
+- ```images``` table
+  new field: ```backup_id``` (foreign key ```bakcups:id```)
+  new field: ```backup_url``` (nbd url used in image transfer ticket)
+
+
+### REST
+
+#### Finding disks enabled for incremental backup
+
+XXX Write me
+
+#### Starting backup
+
+XXX Missing way to specify checkpoint cleanup after backup.
+
+Start incremental backup since backup id "previous-backup-uuid".
+
+Request:
+```
+POST /vms/vm-uuid/backups
+
+<backup>
+    <incremental>previous-backup-uuid</incremental>
+    <disks>
+        <disk id="disk-uuid" />
+        ...
+    </disks>
+</backup>
+```
+
+Response:
+```
+<backup id="backup-uuid">
+    <incremental>previous-backup-uuid</incremental>
+    <disks>
+        <disk id="disk-uuid" />
+        ...
+        ...
+    </disks>
+    <status>initiailizing</status>
+    <creation_date>...
+</backup>
+```
+
+#### Getting backup info
+
+When backup status is "ready", you can start downloading the disks.
+
+Request:
+```
+GET /vms/vm-uuid/backups/backup-uuid
+```
+
+Response:
+```
+<backup id="backup-uuid">
+    <incremental>previous-backup-uuid</incremental>
+    <disks>
+        <disk id="disk-uuid">
+            <image_id>image-uuid</image_id>
+        </disk>
+        ...
+    </disks>
+    <status>ready</status>
+    <creation_date>...
+</backup>
+```
+
+#### Finalizing backup
+
+```
+POST /vms/vm-uuid/backups/backup-uuid/finalize
+
+<action></action>
+```
+
+#### Canceling backup
+
+```
+POST /vms/vm-uuid/backups/backup-uuid/cancel
+
+<action></action>
+```
+
+#### Creating image transfer for incremental restore
+
+XXX Write me
+
+### Incremental backup ticket example
+
+For running VM, qemu will serve the disk using NBD:
+```
+{
+    "url": "nbd:localhost:1234:exportname=/sda"
+}
+```
+
+For stopped VM, we will run one qemu-nbd instance per disk, using unix
+socket:
+```
+{
+    "url": "nbd:unix:/run/vdsm/nbd/xxxyyy-nbd.sock:exportname=/sda"
+}
+```
+
```

Comment:
We need an examples section, showing the API calls for common flows: full backup, incremental backup, falling back to full backup when one disk has no change list, etc.